core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1856,6 +1895,65 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
         saveNow();
     }
 
+    /**
+     * Finds if tx is NOT spending other txns which are in the specified confidence type
+     * @param tx
+     * @param confidenceType
+     * @return true if tx is NOT spending other txns which are in the specified confidence type. false otherwise
+     */
+    private boolean isNotSpendingTxnsInConfidenceType(Transaction tx, ConfidenceType confidenceType) {
+        boolean isNotSpendingTxnsInConfidenceType = true;
+        for (TransactionInput txInput : tx.getInputs()) {
+            Transaction connectedTx = this.getTransaction(txInput.getOutpoint().getHash());
+            if (connectedTx!=null && connectedTx.getConfidence().getConfidenceType().equals(confidenceType)) {
+                isNotSpendingTxnsInConfidenceType = false;
+            }
+        }        
+        return isNotSpendingTxnsInConfidenceType;
+    }
+
+    
+    /**
+     * Creates and returns a new List with the same txns as inputSet but txns are sorted by depencency.
+     * If tx B spends tx A, then tx A should before tx B on the returned List.
+     * Several invocations to this method with the same inputSet could result in lists with txns in different order, 
+     * as there is no guarantee on the order of the returned txns besides what was already stated.
+     * @param inputSet the txns to sort
+     * @return a List of txns where if tx B spends tx A, then tx A should before tx B.
+     */
+    List<Transaction> sortTxnsByDependency(Set<Transaction> inputSet) {
+        List<Transaction> result = new ArrayList<Transaction>(inputSet);
+        for (int i = 0; i < result.size()-1; i++) {
+            boolean txAtISpendsOtherTxInTheList;
+            do {
+                txAtISpendsOtherTxInTheList = false;
+                for (int j = i+1; j < result.size(); j++) {
+                    if (spends(result.get(i), result.get(j))) {
+                        Transaction transactionAtI = result.remove(i);
+                        result.add(j, transactionAtI);
+                        txAtISpendsOtherTxInTheList = true;
+                        break;
+                    }
+                }                        
+            } while (txAtISpendsOtherTxInTheList);
+        }
+        return result;
+    }
+    
+    /**
+     * Finds whether txA spends txB 
+     * @param txA
+     * @param txB
+     * @return true if txA spends txB. False otherwise.
+     */
+    boolean spends(Transaction txA, Transaction txB) {
+        boolean spends = false;
+        for (TransactionInput txInput : txA.getInputs()) {
+            spends |= txInput.getOutpoint().getHash().equals(txB.getHash());
```

Comment:
This can just be "if (txInput.....) return true" right? No need to continue looping after finding that the condition is correct.